### PR TITLE
chore: support ruby 2.4 and use redis in test

### PR DESCRIPTION
**Pull request by TamarinEA**

### CODE REVIEW

File: spec/redis_counters/unique_hash_counter_spec.rb [hunk 1]
```diff
@@ -76,8 +76,8 @@
 
     it { expect(redis.lrange('test_counter_uq:1:partitions', 0, -1)).to eq ['2013-04-28', '2013-04-27'] }
     it { expect(redis.lrange('test_counter_uq:2:partitions', 0, -1)).to eq ['2013-04-27'] }
-    it { expect(redis.smembers('test_counter_uq:1:2013-04-27')).to eq ['4'] }
-    it { expect(redis.smembers('test_counter_uq:2:2013-04-27')).to eq ['3', '2', '1'] }
-    it { expect(redis.smembers('test_counter_uq:1:2013-04-28')).to eq ['5', '1'] }
+    it { expect(redis.smembers('test_counter_uq:1:2013-04-27')).to match_array ['4'] }
+    it { expect(redis.smembers('test_counter_uq:2:2013-04-27')).to match_array ['3', '2', '1'] }
+    it { expect(redis.smembers('test_counter_uq:1:2013-04-28')).to match_array ['5', '1'] }
```
Comment (apress-hound) [inline spec/redis_counters/unique_hash_counter_spec.rb:81]: Style/WordArray: Use %w or %W for an array of words.

File: spec/redis_counters/unique_hash_counter_spec.rb [hunk 2]
```diff
@@ -76,8 +76,8 @@
 
     it { expect(redis.lrange('test_counter_uq:1:partitions', 0, -1)).to eq ['2013-04-28', '2013-04-27'] }
     it { expect(redis.lrange('test_counter_uq:2:partitions', 0, -1)).to eq ['2013-04-27'] }
-    it { expect(redis.smembers('test_counter_uq:1:2013-04-27')).to eq ['4'] }
-    it { expect(redis.smembers('test_counter_uq:2:2013-04-27')).to eq ['3', '2', '1'] }
-    it { expect(redis.smembers('test_counter_uq:1:2013-04-28')).to eq ['5', '1'] }
+    it { expect(redis.smembers('test_counter_uq:1:2013-04-27')).to match_array ['4'] }
+    it { expect(redis.smembers('test_counter_uq:2:2013-04-27')).to match_array ['3', '2', '1'] }
```
Comment (apress-hound) [inline spec/redis_counters/unique_hash_counter_spec.rb:80]: Style/WordArray: Use %w or %W for an array of words.

File: spec/redis_counters/hash_counter_spec.rb [hunk 3]
```diff
@@ -407,8 +401,7 @@
         before { counter.delete_partitions!(partition_2) }
 
         it { expect(counter.partitions).to have(2).row }
-        it { expect(counter.partitions.first).to eq partition_1 }
-        it { expect(counter.partitions.last).to eq partition_3 }
+        it { expect(counter.partitions).to match_array [partition_1, partition_3]  }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:404]: Layout/ExtraSpacing: Unnecessary spacing detected.

File: spec/redis_counters/hash_counter_spec.rb [hunk 4]
```diff
@@ -396,9 +392,7 @@
         before { counter.delete_partitions!(:param1 => 21, &error_proc) rescue nil }
 
         it { expect(counter.partitions).to have(3).row }
-        it { expect(counter.partitions.first).to eq partition_1 }
-        it { expect(counter.partitions.second).to eq partition_2 }
-        it { expect(counter.partitions.last).to eq partition_3 }
+        it { expect(counter.partitions).to match_array [partition_1, partition_2, partition_3]  }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:395]: Layout/ExtraSpacing: Unnecessary spacing detected.

File: spec/redis_counters/hash_counter_spec.rb [hunk 5]
```diff
@@ -375,6 +370,7 @@
         before { counter.delete_partitions!(partition_2) }
 
         it { expect(counter.partitions).to have(2).row }
+        it { expect(counter.partitions).to match_array [partition_1, partition_3]  }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:373]: Layout/ExtraSpacing: Unnecessary spacing detected.

File: spec/redis_counters/hash_counter_spec.rb [hunk 6]
```diff
@@ -182,17 +182,14 @@
...
+      it { expect(counter.partitions).to match_array [partition_1, partition_2, partition_3]  }
     end
 
     context 'when one partition param given' do
       it { expect(counter.partitions(:param1 => 11)).to have(1).partition }
       it { expect(counter.partitions(:param1 => 11).first).to eq partition_1 }
       it { expect(counter.partitions(:param1 => 21)).to have(2).partition }
-      it { expect(counter.partitions(:param1 => 21).first).to eq partition_2 }
-      it { expect(counter.partitions('param1' => 21).second).to eq partition_3 }
+      it { expect(counter.partitions(:param1 => 21)).to match_array [partition_2, partition_3]  }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:192]: Layout/ExtraSpacing: Unnecessary spacing detected.

File: spec/redis_counters/hash_counter_spec.rb [hunk 7]
```diff
@@ -182,17 +182,14 @@
 
     context 'when no partition params given' do
       it { expect(counter.partitions).to have(3).partition }
-      it { expect(counter.partitions.first).to eq partition_1 }
-      it { expect(counter.partitions.second).to eq partition_2 }
-      it { expect(counter.partitions.third).to eq partition_3 }
+      it { expect(counter.partitions).to match_array [partition_1, partition_2, partition_3]  }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:185]: Layout/ExtraSpacing: Unnecessary spacing detected.

File: spec/redis_counters/hash_counter_spec.rb [hunk 8]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 describe RedisCounters::HashCounter do
-  let(:redis) { MockRedis.new }
+  let(:redis) {Redis.current }
```
Comment (apress-hound) [inline spec/redis_counters/hash_counter_spec.rb:4]: Layout/SpaceInsideBlockBraces: Space missing inside {.

**Review by isqad — APPROVED** [review]

**Review by c1n1c — APPROVED** [review]

**Pull request merged**